src/Language/PureScript/AST/Declarations.hs

```diff
@@ -154,7 +154,7 @@ data Declaration
   -- A type instance declaration (name, dependencies, class name, instance types, member
   -- declarations)
   --
-  | TypeInstanceDeclaration Ident [Constraint] (Qualified ProperName) [Type] [Declaration]
+  | TypeInstanceDeclaration Ident [Constraint] (Qualified ProperName) [Type] (Maybe [Declaration])
```

Comment:
I think I'd prefer to reuse it, but to use a sum type inside in place of `Maybe`. That way, we can extend it to `newtype` deriving easily later, and it's still clear.